### PR TITLE
Pydantic V2 support

### PR DESCRIPTION
**Pull request by ansenlong**
Added Pydantic V2 support

### CODE REVIEW

**Review by fcurella — CHANGES_REQUESTED** [review]
Thank you! Could you add some tests and run `make lint` to fix the linting issues?

File: datauri/__init__.py [hunk 1]
```diff
@@ -169,6 +169,16 @@ def validate(cls, v: str) -> Self:
             raise ValueError("invalid data-uri format")
         return m
 
+    @classmethod
+    def __get_pydantic_json_schema__(cls, core_schema: Any, handler: Any) -> Any:
```
Comment (fcurella) [inline datauri/__init__.py:173]: shouldn't `core_schema` have type `Mapping[str, Any]`, as in [pydantic_core](https://github.com/pydantic/pydantic-core/blob/main/python/pydantic_core/core_schema.py#L3855C29-L3855C46)?

File: tests/test_pydantic.py [hunk 2]
```diff
@@ -4,6 +4,13 @@
 
 pydantic = pytest.importorskip("pydantic")
 
+try:
+    pydantic.version.version_info()
+    func_json = "model_dump_json"
+except:
```
Comment (fcurella) [inline tests/test_pydantic.py:10]: can we be explicit in what exception we are trying to catch? Is it `AttributeError`?

File: tests/test_pydantic.py [hunk 3]
```diff
@@ -4,6 +4,13 @@
...
+    pydantic.utils.version_info()
```
Comment (fcurella) [inline tests/test_pydantic.py:11]: what's the point of calling `version_info()` again?

File: tests/test_pydantic.py [hunk 4]
```diff
@@ -13,8 +20,7 @@ class Model(pydantic.BaseModel):
     instance = Model(content=t)
     assert isinstance(instance.content, DataURI)
     assert (
-        instance.json()
-        == '{"content": "data:text/plain;charset=utf-8;base64,VGhlIHF1aWNrIGJyb3duIGZveCBqdW1wZWQgb3ZlciB0'
-        'aGUgbGF6eSBkb2cu"}'
+        instance.__getattr__(func_json)()
+        == '{"content":"data:text/plain;charset=utf-8;base64,VGhlIHF1aWNrIGJyb3duIGZveCBqdW1wZWQgb3ZlciB0aGUgbGF6eSBkb2cu"}'
```
Comment (fcurella) [inline tests/test_pydantic.py:24]: could we test both?

File: tests/test_pydantic.py (hunk 3 above)
Comment (ansenlong) [reply tests/test_pydantic.py]: removed

File: tests/test_pydantic.py (hunk 2 above)
Comment (ansenlong) [reply tests/test_pydantic.py]: added

File: datauri/__init__.py (hunk 1 above)
Comment (ansenlong) [reply datauri/__init__.py:173]: Yes, added mapping Dict

File: tests/test_pydantic.py (hunk 4 above)
Comment (ansenlong) [reply tests/test_pydantic.py:25]: I don't know how to test for this edge case
